examples/solvers/chambolle_pock_denoising.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Denoising using Chambolle-Pock solver."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import numpy as np
+import scipy
+import scipy.ndimage
+import matplotlib.pyplot as plt
+
+# Internal
+import odl
+
+
+# TODO: Use BroadCastOperator instead of ProductSpaceOperator
+
+# Read the image
+image = np.rot90(scipy.misc.ascent()[::2, ::2], 3).astype('float')
+shape = image.shape
+# Rescale max to 1
+image /= np.max(image[:])
+
+# Discretized spaces
+discr_space = odl.uniform_discr([0, 0], shape, shape)
+
+# Original image
+orig = discr_space.element(image)
+
+# Add noise and rescale back to [0, 1]
+image += np.random.normal(0, 0.1, shape)
+image = (image - np.min(image)) / (np.max(image) - np.min(image))
```

Comment:
I dislike rescaling an image according to min/max after we add normal noise, this will in the limit of infinitely many pixels give an image with only noise. Perhaps only rescale before adding noise?